### PR TITLE
MDv3: get rid of negative ephemeral pubkey index hack

### PR DESCRIPTION
**Pull request by aalness**

### CODE REVIEW

**Comment by aalness** [comment]
Ok to review, circleci failure will be fixed by: https://github.com/keybase/kbfs/pull/465

**Comment by aalness** [comment]
Testing note: I manually enabled MDv3 and verified all `TestKeyManager*` tests passed.

File: libkbfs/bare_root_metadata_v3.go [hunk 1]
```diff
@@ -573,19 +575,17 @@ func (md *BareRootMetadataV3) GetTLFCryptKeyParams(
 			TLFCryptKeyServerHalfID{}, false, nil
 	}
 
-	var index int
 	var publicKeys kbfscrypto.TLFEphemeralPublicKeys
 	var keyType string
-	if info.EPubKeyIndex >= 0 {
```
Comment (jzila) [inline libkbfs/bare_root_metadata_v3.go]: this still needs to work with old metadata...

Comment (aalness) [inline libkbfs/bare_root_metadata_v3.go]: `BareRootMetadataV2` still uses the hack in its implementation.

Comment (jzila) [inline libkbfs/bare_root_metadata_v3.go]: Have we been writing new metadata at all yet?

Comment (aalness) [inline libkbfs/bare_root_metadata_v3.go]: It's not enabled in production nor staging yet.

**Pull request merged**